crates/ruma-identifiers/src/matrix_uri.rs

```diff
@@ -28,35 +28,112 @@ const TO_ENCODE: &AsciiSet = &CONTROLS
     .add(b';')
     .add(b'=');
 
-/// A reference to a user, room or event.
+/// All Matrix Identifiers that can be represented as a Matrix URI.
+#[derive(Debug, PartialEq, Eq)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
```

Comment:
I don't think we should allow exhaustive matching on this, even with the unstable feature (that really just exists for Conduit at this point).